atlp-rwanda/destructors-ec-bn
# #184637137 Reset Password via email

#### What does this PR do?

This PR is API for Resetting  the password of the user via email in case he/she forget the password he was using. 

#### Description of Task to be completed?

Its' invaluable in instances where our users forget their login credentials.

#### How should this be manually tested?

- clone the repository 
- checkout ft-reset-password-184637137 branch
- to install the dependencies and devdependencies used run `npm install` in terminal
- check .env.example to see the secret key needed to be in .env file
- create .env file depending on .env.example file
- to run server run `npm run dev` in terminal and check if server is working
- then open your browser and in URL field enter localhost:PORT/api-docs to open swagger UI
- Under RESET PASSWORD, the first field of  POST   enter the email of user you provided during registration then hint execute
- then you will receive email containing reset password link 
-  Under PATCH field use the token return in the response body under in POST field and add it in token field provided in parameters
- in the request body enter password and confirmPassword as shown  and then hit execute
- you will see a message  saying password changed successfully in response body 

#### Any background context you want to provide?

#### What are the relevant pivotal tracker/Trello stories?

- tracker : #184637137

#### Screenshots (if appropriate)

#### Questions:
